### PR TITLE
Update to Lensv2 APIs

### PR DESCRIPTION
**Pull request by HashWarlock**
## Description
Updates to the LensAPI Oracle to use `viem`, remove package deps, update APIs to use Lensv2 and update queries.

### CODE REVIEW

**Review by pacoyang — COMMENTED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ yarn build-function
 
 To simulate the expected result locally, run the Phala Oracle function now with this command:
 ```bash
-yarn run-function -a 0x0000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000043078303100000000000000000000000000000000000000000000000000000000 https://api-mumbai.lens.dev
+yarn run-function -a 0x0000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000043078303100000000000000000000000000000000000000000000000000000000 https://api-v2.lens.dev/
```
Comment (shelvenzhou) [inline README.md]: shall we use `https://api-v2-mumbai-live.lens.dev/` here?

File: README.md [hunk 2]
```diff
@@ -84,20 +84,23 @@ yarn run-function -a 0x000000000000000000000000000000000000000000000000000000000
 <details>
   <summary><u>How the query looks under the hood</u></summary>
 
-- HTTP Endpoint: https://api-mumbai.lens.dev
+- HTTP Endpoint: https://api-v2.lens.dev
```
Comment (shelvenzhou) [inline README.md]: same above.

File: package.json [hunk 3]
```diff
@@ -23,8 +26,8 @@
     "localhost-deploy": "hardhat run --network localhost ./scripts/localhost/deploy.ts",
     "localhost-watch": "phat-fn watch",
     "localhost-push-request": "hardhat run --network localhost ./scripts/localhost/push-request.ts",
-    "test-deploy-function": "phat-fn upload --coreSettings=https://api-mumbai.lens.dev/",
-    "main-deploy-function": "phat-fn upload --mode=production --coreSettings=https://api.lens.dev/",
+    "test-deploy-function": "phat-fn upload --coreSettings=https://api-v2-mumbai.lens.dev/",
```
Comment (shelvenzhou) [inline package.json]: use `https://api-v2-mumbai-live.lens.dev/`

File: src/index.ts [hunk 4]
```diff
@@ -109,37 +47,42 @@ function fetchLensApiStats(lensApi: string, profileId: string): any {
     "User-Agent": "phat-contract",
```
Comment (shelvenzhou) [inline src/index.ts:63]: Based on https://api-v2-docs.lens.xyz/, you need to set `user-agent: spectaql` here. But I haven't tested yet.

File: README.md (hunk 2 above)
Comment (HashWarlock) [reply README.md]: I can switch to mumbai for testnet. I was using mainnet lens v2 for example to show more data sense the mumbai testnet has little to no stats returned

File: src/index.ts (hunk 4 above)
Comment (HashWarlock) [reply src/index.ts:63]: The sepctaql doesn't seem to affect queries based on tests

File: package.json [hunk 5]
```diff
@@ -3,6 +3,9 @@
   "version": "0.0.1",
```
Comment (Leechael) [inline package.json]: Let's bump the version to 1.0.0

**Review by Leechael — APPROVED** [review]
lgtm

File: package.json (hunk 5 above)
Comment (HashWarlock) [reply package.json]: Done

**Pull request merged**